sdk/js/src/relayer/relayer/helpers.ts

```diff
@@ -348,13 +318,163 @@ export function vaaKeyToVaaKeyStruct(vaaKey: VaaKey): VaaKeyStruct {
   };
 }
 
+export async function getWormholeRelayerInfoByHash(
+  deliveryHash: string,
+  targetChain: ChainName,
+  sourceChain: ChainName | undefined,
+  sourceVaaSequence: number | undefined,
+  infoRequest?: InfoRequestParams
+): Promise<DeliveryTargetInfo[]> {
+  const environment = infoRequest?.environment || "MAINNET";
+  const targetChainProvider =
+    infoRequest?.targetChainProviders?.get(targetChain) ||
+    getDefaultProvider(environment, targetChain);
+
+  if (!targetChainProvider) {
+    throw Error(
+      "No default RPC for this chain; pass in your own provider (as targetChainProvider)"
+    );
+  }
+  const targetWormholeRelayerAddress =
+    infoRequest?.wormholeRelayerAddresses?.get(targetChain) ||
+    getWormholeRelayerAddress(targetChain, environment);
+  const wormholeRelayer = getWormholeRelayer(
+    targetChain,
+    environment,
+    targetChainProvider,
+    targetWormholeRelayerAddress
+  );
+
+  const blockNumberSuccess = await wormholeRelayer.deliverySuccessBlock(
+    deliveryHash
+  );
+  const blockNumberFailure = await wormholeRelayer.deliveryFailureBlock(
+    deliveryHash
+  );
+  const blockNumber = blockNumberSuccess.gt(0)
+    ? blockNumberSuccess
+    : blockNumberFailure;
+
+  if (blockNumber.toNumber() === 0) return [];
+
+  const blockRange =
+    infoRequest?.targetBlockRange ||
+    (targetChain === "arbitrum"
```

Comment:
y?

Reply:
There is weirdness with arbitrum where if you call 'block.number', it gives you the L1 block number (the ethereum one) - and this is what is stored in the 'replay protection mapping' - so basically that value isn't useful in finding the delivery here

Reply:
you mind writing that out as a comment here?